Yosemite/Yosemite/Tools/POS/POSOrderService.swift

```diff
@@ -0,0 +1,219 @@
+    let itemID: Int64?
+    let product: POSItem
+    let quantity: Decimal
+
+    public init(itemID: Int64?, product: POSItem, quantity: Decimal) {
+        self.itemID = itemID
+        self.product = product
+        self.quantity = quantity
+    }
+}
+
+public struct POSOrder {
+    public let siteID: Int64
+    public let orderID: Int64
+    public let total: String
+    public let totalTax: String
+    public let currency: String
+    let items: [POSOrderItem]
+}
+
+extension POSOrder {
+    init(order: Order) {
+        self.init(siteID: order.siteID,
+                  orderID: order.orderID,
+                  total: order.total,
+                  totalTax: order.totalTax,
+                  currency: order.currency,
+                  items: order.items.map { POSOrderItem(orderItem: $0) })
+    }
+}
+
+struct POSOrderItem {
+    let itemID: Int64
+
+    /// The product ID of a product order item, or the ID of the variable product if the order item is a product variation.
+    let productID: Int64
+    let quantity: Decimal
+
+    func toOrderItem() -> OrderItem {
+        .init(itemID: itemID,
+              name: "",
+              productID: productID,
+              variationID: .zero,
+              quantity: quantity,
+              price: .zero,
+              sku: nil,
+              subtotal: "",
+              subtotalTax: "",
+              taxClass: "",
+              taxes: [],
+              total: "",
+              totalTax: "",
+              attributes: [],
+              addOns: [],
+              parent: nil,
+              bundleConfiguration: [])
+    }
+
+    init(orderItem: OrderItem) {
+        self.itemID = orderItem.itemID
+        self.productID = orderItem.productID
+        self.quantity = orderItem.quantity
+    }
+}
+
+public protocol POSOrderServiceProtocol {
+    /// Syncs order based on the cart.
+    /// - Parameters:
+    ///   - cart: Cart with optional items (product & quantity).
+    ///   - order: Optional latest remotely synced order. Nil when syncing order for the first time.
+    ///   - allProducts: Necessary for removing existing order items with products that have been removed from the cart.
+    /// - Returns: Order from the remote sync.
+    func syncOrder(cart: [POSCartItem], order: POSOrder?, allProducts: [POSItem]) async throws -> POSOrder
+
+    /// Updates status of an order and syncs it
+    /// - Parameters:
+    ///   - posOrder: POS order.
+    ///   - status: New order status.
+    /// - Returns: Updated and synced POS order.
+    func updateOrderStatus(posOrder: POSOrder, status: OrderStatusEnum) async throws -> POSOrder
+
+    /// Creates WOO Order from POS Order.
+    /// - Parameters:
+    ///   - posOrder: POS order.
+    /// - Returns: Order created from posOrder data.
+    func order(from posOrder: POSOrder) -> Order
+
+    /// Creates an empty autodraft order
+    func createAutoDraftOrder() -> Order
```

Comment:
❓ Is this function necessary in the protocol that the POS app layer interacts with? From my understanding, this is an implementation detail in the Yosemite layer to create the initial order. The following code where this is used

```swift
        if let posOrder {
            initialOrder = order(from: posOrder)
        }
        else {
            initialOrder = createAutoDraftOrder()
        }
```

can be moved to a separate private function like `createInitialOrder` that takes in an optional POS order.